src/inmanta/data/model.py

```diff
@@ -459,10 +459,22 @@ class PagingBoundaries:
     Represents the lower and upper bounds that should be used for the next and previous pages
     when listing domain entities.
 
-    :param start: largest value of the page for the primary sort column.
-    :param end: smallest value of the page for the primary sort column.
-    :param first_id: largest value of the page for the secondary sort column, if there is one.
-    :param last_id: smallest value of the page for the secondary sort column, if there is one.
+    The largest / smallest value of the current page represents respectively the min / max boundary value (exclusive) for the
+    neighbouring pages. Which represents next and which prev depends on sorting order (ASC or DESC).
+
+    Boundary values represent max and min values, regardless of sorting direction (ASC or DESC), i.e.
+    - ASC sorting based on next links: (>) end,last_id
+    - ASC sorting based on prev links: (<) start,first_id
+    - DESC sorting based on next links: (<) start,first_id
+    - DESC sorting based on prev links: (>) end,last_id
+
+    So, while the names "start" and "end" might seem to indicate "left" and "right" of the page, they actually mean "highest"
+    and "lowest".
```

Comment:
While trying to understand this I added a tangible example with 2 columns (int, letter) that shows which param is taken into consideration on which page depending on the sorting order. It also shows the inclusive/exclusive ie what is on which page. Is this accurate ? (Feel free to ignore)
```suggestion
    and "lowest".
    
                                                                                                                                     
                  |        prev             |                current page                    |    next                           
     -------------|-------------------------|------------------------------------------------|----------------------             
     ASC sorting  | [  ...  ] (100 c)       | [ (100 c) ( 100 d) ... (10 000 r) (10 000 s) ] | (10 000 s) [ ...   ]              
                  |            start = 100  |     end = 100            start = 10 000        |     end = 10 000                  
                  |            first_id = c |     last_id = c          first_id = s          |     last_id = s                   
                  |                                                                                                              
     DESC sorting | [  ...  ] (10 000 s)    | [ (10 000 s) (10 000 r) ... (100 d) ( 100 c) ] | (100 c) [ ...   ]                 
                  |            end = 10 000 |          start = 10000          end = 100      |     start = 100                   
                  |            last_id = s  |          first_id = s           last_id = c    |     first_id = c                  
                                                                                                                                 
```

Reply:
I adjusted some values, but it's way cleaner, thank you!